src/main/java/de/zalando/aruha/nakadi/partitioning/UserDefinedPartitioningStrategy.java

```diff
@@ -0,0 +1,31 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.PartitioningException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.List;
+
+public class UserDefinedPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+            throws PartitioningException {
+        try {
+            final String partition = event.getJSONObject("metadata").getString("partition");
```

Comment:
Would you throw an exception in case it happens? Because if so I would say the behaviour would be the same, since `getJSONObject` also throws an exception in case of null.